crypto/bio/bio_lib.c

```diff
@@ -856,21 +863,21 @@ int BIO_connect_retry(BIO *bio, int timeout)
 
```

Comment:
I wonder if I look at this code above
```C
    if (!blocking)
        BIO_set_nbio(bio, 1);
```

is that garanteed that the bio is always in blocking mode so
wouldn't it be safer to call `BIO_set_nbio(bio, !blocking);` here ?

Reply:
Good question. I had inferred that from programming examples.
Looking at the documentation of `BIO_set_nbio()` confirms this:
```
Blocking I/O is the default.
```

Reply:
Yes, unless the BIO is initialized otherwise, and when you say,
the blocking sockets return with BIO_should_retry that makes me
curious, what triggers this behavior.

Reply:
Would be great if you anyone else with more network programming experience than me could investigate into that. Shall we open an issue?

Reply:
If you can setup something reproducible, yes, why not.
Currently it looks foggy to me.

Reply:
All right, then I won't open an issue now because I currently cannot reproduce the crash,
and I think the newly added TODO should be sufficient for now.